### DISCUSSION TITLE
The Windows build has been failing since May 1st.

### DISCUSSION THREAD

**Discussion by aitaro** [codemagic-ci-cd/codemagic-docs#2501] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Run `flutter build windows` on windows machine.

### Expected results

The build succeeds.

### Actual results

It fails with the following error message:

```
Nuget is not installed.
Attempting to download nuget.
CMake Deprecation Warning at C:/Users/builder/clone/dart/apps/mobile/build/windows/x64/extracted/firebase_cpp_sdk_windows/CMakeLists.txt:17 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


Nuget.exe not found, trying to download or use cached version.
Package 'Microsoft.Windows.CppWinRT 2.0.210806.1' is not found in the following primary source(s): 'C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\'. Please verify all your online package sources are available (OR) package id, version are specified correctly.
CMake Error at flutter/ephemeral/.plugin_symlinks/permission_handler_windows/windows/CMakeLists.txt:29 (message):
  Failed to install nuget package Microsoft.Windows.CppWinRT.2.0.210806.1


Building Windows application...                                    27.8s
Unable to generate build files


Build failed :|
Step 7 script `Build Windows` exited with status code 1
```

When I ran a build with the same tag two days ago, the build was successful, but when I ran it today, it failed.
succeed build id: 66314e7891bea5d3aed75773
failed build id: 66333f7efe6c8361fb8584cf

I noticed Java version was changed.
prev:
```
== Use Java version 11 ==
> java -version
openjdk version "11.0.13" 2021-10-19
OpenJDK Runtime Environment 18.9 (build 11.0.13+8)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.13+8, mixed mode)

== Codemagic CLI tools ==
> codemagic-cli-tools.BAT version
codemagic-cli-tools 0.50.7
```
now:
```
== Use Java version 11 ==
> java -version
openjdk version "11.0.22" 2024-01-16
OpenJDK Runtime Environment Temurin-11.0.22+7 (build 11.0.22+7)
OpenJDK 64-Bit Server VM Temurin-11.0.22+7 (build 11.0.22+7, mixed mode)

== Codemagic CLI tools ==
> codemagic-cli-tools.BAT version
codemagic-cli-tools 0.50.7
```

### Build id (optional)

66333f7efe6c8361fb8584cf

**Comment by mohammedbabelly20** [comment]
Hey @aitaro 
The Windows base image has bee upgraded yesterday (see [announcement](https://github.com/orgs/codemagic-ci-cd/discussions/2474))
We already applied the first fix for the Nuget issues, and we're keeping an eye on any coming issues.

**Reply by aitaro** [reply]
Thank you for the explanation. Upon further investigation, I discovered that the error occurs in the permission_handler_windows package (which is a commonly used package in Flutter).

The issue I'm encountering is exactly the same as the one described in this GitHub issue: https://github.com/Baseflow/flutter-permission-handler/issues/819. According to this comment (https://github.com/Baseflow/flutter-permission-handler/issues/819#issuecomment-1673271795), the problem might be related to incorrect settings in the NuGet.Config file. Is it possible to modify the NuGet.Config settings on the Codemagic Windows instance to address this issue?

**Reply by mohammedbabelly20** [reply]
This issue has been addressed in the last base image update, can you try running your builds again?
